### ISSUE TITLE
Download in podboat fails because filename is too long

### ISSUE THREAD

**Issue by edi9999** [newsboat/newsboat#1208]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
download-path "/home/edgar/Podcasts/%n"
podlist-format "%t %T %N %4i [%6dMB/%6tMB] [%5p %%] [%7k kb/s] %-20S -> %F"
download-filename-format "%N-%F - %t.%e"
```

Steps to reproduce the issue:

1. With just the following feed : https://feeds.backtracks.fm/indiehackers/indie-hackers-podcast/feed.xml

2. When enqueuing the podcast #174, `From Millions in Revenue to Staring Bankruptcy in the Face with Aline Lerner of Interviewing.io.mp3`

3. Then opening podboat to download the file

I get the status "failed" for the download.

From my investigation, this happens because the filename is too long for the system. ie I can reproduce the same issue if I try to do : 

```
touch 'The Indie Hackers Podcast-2020-07-09 #174 – From Millions in Revenue to Staring Bankruptcy in the Face with Aline Lerner of Interviewing.io.mp3'
```

Ideally, I would be able to change the name of the file when such a failure occurs manually.

It could also be possible to drop characters at the end until the write succeeds.

What do you think ?

**Comment by edi9999** [comment]
At least showing the reason for the error would be better than to just mark the status as failed.

**Comment by Minoru** [comment]
Thank you for the report! Podboat doesn't get as much developer attention as Newsboat proper, so I have no idea if/when this would be fixed. PRs welcome!

> Ideally, I would be able to change the name of the file when such a failure occurs manually. It could also be possible to drop characters at the end until the write succeeds.

Either approach sound acceptable to me. I leave the decision to whoever wants to implement this.

Note though that Podboat operates on a full *path*, not just a filename. If shortening the filename is not enough, it might be necessary to shorten the names of intermediate directories as well. (*Hint*, *hint*: Rust has handy `Path` and `PathBuf` types that make writing this sort of code easier than in C++; so consider putting the "meat" of the algorithm into the libnewsboat crate.)

> At least showing the reason for the error would be better than to just mark the status as failed.

Definitely. This enhancement doesn't depend on manual/automatic renaming, so I'll turn it into a separate issue — perhaps it'll be implemented sooner then.

**Comment by edi9999** [comment]
Just so that you know, the limitation on my system is 140 characters because the folder is in an encrypted ecryptfs partition : https://unix.stackexchange.com/a/32834

You might not be able to reproduce on an unencrypted fs, since usually the limit is higher (255 characters).

**Comment by dennisschagt** [comment]
It looks like this kind of issue is now detected and shown in Podboat's status line (introduced in https://github.com/newsboat/newsboat/pull/1256):
![image](https://user-images.githubusercontent.com/4629607/149616081-7cb7d298-e2aa-44c9-b6af-d10445e310bb.png)

**Comment by Minoru** [comment]
Right! And the user can edit the filename in the queue file to change the length. I'll keep the issue open because I still think Podboat (or Newsboat, who generates the names) by shortening the names.
